features/users/operations.py

```diff
@@ -230,3 +297,224 @@ def remove_user_from_role(user_id: int, role_id: int) -> None:
 
         session.add(user)
         session.commit()
+
+
+def _prepare_mail_template(*, token_type: str, token: str, recipient: str):
+    """
+    Prepare email template
+
+    :param token_type:
+    :param token:
+    :param recipient:
+    :return:
+    """
+
+    config = configuration.Config()
+
+    templates = Jinja2Templates('features/users/templates')
+    template_path = 'confirmation-email-template.html' if token_type == TokenTypes.EMAIL_CONFIRMATION \
+        else 'password-reset-email.html'
+    template = templates.get_template(template_path)
+
+    confirmation_link = f'{config.server.host}:{config.server.port}/users/confirm-email/{token}'
+
+    html_content = template.render(
+        recipient_name=recipient,
+        token_confirmation_link=confirmation_link,
+    )
+
+    return html_content
+
+
+async def _send_mail(*, token_type: str, recipient_email: str, username: str, html_content):
+    """
+    Create headers, payload and send email
+
+    :param token_type:
+    :param recipient_email:
+    :param username:
+    :param html_content:
+    :return:
+    """
+    brevo = configuration.BrevoSettings()
+    api_url = brevo.email_api_url
+    headers = {
+        "Content-Type": "application/json",
+        "api-key": brevo.email_api_key,
+        "Accept": "application/json",
+    }
+
+    payload = {
+        "sender": {"name": brevo.email_sender, "email": brevo.email_from},
+        "to": [{"email": recipient_email, "name": username}],
+        "subject": features.users.constants.EMAIL_SUBJECT[token_type],
+        "htmlContent": html_content,
+    }
+
+    async with AsyncClient() as client:
+        response = await client.post(api_url, headers=headers, json=payload)
+
+        if response.status_code != 201:
+            raise HTTPException(
```

Comment:
from operations you should not know for fastpi. Create a custom error and raise it. Catch it in the router and re raise it as httpException with status code for failed dependencie